### ISSUE TITLE
Please create separate landing pages per Python version

### ISSUE THREAD

**Issue by gvanrossum** [python/pythondotorg#1693]
In Azure we support specific Python feature releases (currently 3.6, 3.7, 3.8). Users have their reasons to stick to a particular version. But we don't want to give them a choice about the bugfix release -- if they want 3.8, they should get the latest 3.8.x release (currently 3.8.6). In our docs (e.g. [here](https://docs.microsoft.com/en-us/azure/azure-functions/create-first-function-cli-python?tabs=azure-cli%2Ccmd%2Cbrowser#configure-your-local-environment)) we'd like to link directly to the download page for each version.  But here we have a slight problem, in that there is no landing page for "the latest 3.8 release" (that I know of or can find). So our doc team ended up linking to 3.8.2 (which was the latest at the time they wrote the docs) and never updated the link to point to 3.8.6.

On python.org we have a landing page for "[the latest version](https://www.python.org/downloads/)" (currently 3.9.0) and "[a specific version](https://www.python.org/downloads/release/python-386/)" (e.g. 3.8.2). Could you also add a landing page for "the latest bugfix release for 3.x"? E.g. for 3.8 this would currently refer the user to 3.8.6, but once 3.8.7 is released it would automatically point there. I'd prefer this not to be a redirect but just a page with some text that says e.g. "the latest version of 3.8 is 3.8.6, download it here". Ideally it would also list previous versions -- especially important when a release goes into maintenance mode, since in most cases users would rather download the latest binary release than attempt to download and build the latest source-only release themselves.

**Comment by Jackenmen** [comment]
So there already exists `https://python.org/downloads/latest/python3` that redirects to latest stable Python 3. What kind of URL would you want to use for version-specific redirects?

I was thinking that maybe the implementation could URL such as `https://python.org/downloads/latest/python3/3.9/`, would that be acceptable?

**Comment by Jackenmen** [comment]
Oh and I just remembered that security releases are source-only (which now that I re-read the description, I see was mentioned, sorry about that...) so yeah, we would also need a redirect that redirects to the latest bugfix release of `3.x` that has binaries. Not sure what URL pattern should be used for that.

This doesn't really solve the whole issue as it would just be a simple redirect, not a landing page for a specific `3.x`.
Nonetheless, I think it would still be helpful to have such redirects as it should be simpler to implement until someone in the community steps up and contributes a full landing page view *and* it does solve the problem of not being able to link to the latest bugfix version of specific `3.x` version.

**Comment by gvanrossum** [comment]
Yeah, a redirect would be fine, or at least a step in the right direction. For purely esthetic reasons I'd rather not have an extra directory level in there -- I'd rather have .../python3.9/ than .../python3/3.9/ but if that's technically difficult I don't care that much.

And yes, it would be nice if it directed to the last release with binaries. In the past we used to have a note in each landing page pointing to the most recent landing page, but I think we either stopped doing that or it always goes to the latest version, which is not what I want. Maybe the last release with binaries could have a link at the top to the latest source-only release if there is one, but for this particular use case it doesn't matter much.

**Comment by hugovk** [comment]
Please see PR https://github.com/python/pythondotorg/pull/2816.
